web/federate.go

```diff
@@ -74,7 +74,7 @@ func (h *Handler) federation(w http.ResponseWriter, req *http.Request) {
 	}
 	sort.Sort(byName(vector))
 
-	externalLabels := h.externalLabels.Clone()
+	externalLabels := h.config.GlobalConfig.ExternalLabels.Clone()
```

Comment:
@brancz Instead of having [`configString`](https://github.com/prometheus/prometheus/pull/2711/files#diff-73cbd2009b0e4f156e8ca0f47e95b016L76) and [`externalLabels`](https://github.com/prometheus/prometheus/pull/2711/files#diff-73cbd2009b0e4f156e8ca0f47e95b016L82) in the `Handler` struct, I propose with this PR just having the [`config`](https://github.com/prometheus/prometheus/pull/2711/files#diff-73cbd2009b0e4f156e8ca0f47e95b016R76) itself in there. Whenever we need `externalLabels` and `configString` we can get it from `Handler.config.String()` and `Handler.config.GlobalConfig.ExternalLabels`.

I find this cleaner. In addition it enables us to access more than just the external labels and the config string in both *web.go* and *api.go* in the future. @brancz What do you think?